api/v1beta1/activemqartemis_types.go

```diff
@@ -377,6 +377,16 @@ type DeploymentPlanType struct {
 	// Specifies the container security context
 	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Container Security Context"
 	ContainerSecurityContext *corev1.SecurityContext `json:"containerSecurityContext,omitempty"`
+	// Specifies additional Volumes to be attached to the broker pods
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Extra Volumes"
+	ExtraVolumes []corev1.Volume `json:"extraVolumes,omitempty"`
+	// Specifies mount options for extraVolumes
+	// +optional
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Extra Mount Options"
+	ExtraVolumeMounts []*corev1.VolumeMount `json:"extraVolumeMounts,omitempty"`
+	// Specifies Extra Volume Claims Templates for the broker pods
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Extra Volume Claims Templates"
+	ExtraVolumeClaimTemplates map[string]corev1.PersistentVolumeClaimSpec `json:"extraVolumeClaimTemplates,omitempty"`
```

Comment:
The field `ExtraVolumeClaimTemplates` could be an array of [PersistentVolumeClaim](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#persistentvolumeclaim-v1-core) as in [statefulset](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#statefulset-v1-apps). In this way the users could easily add the labels and annotations.
```suggestion
	ExtraVolumeClaimTemplates []corev1.PersistentVolumeClaim `json:"extraVolumeClaimTemplates,omitempty"`
```

Reply:
+1

Reply:
@brusdev Hi Dom, I'm having issue after I made this field as list of PersistentVolumeClaim. I cannot make it right, when i deploy such a cr 
`
apiVersion: broker.amq.io/v1beta1
apiVersion: broker.amq.io/v1beta1
kind: ActiveMQArtemis
metadata:
  name: ex-aao
spec:
  deploymentPlan:
    size: 1
    extraVolumeClaimTemplates:
    - apiVersion: v1
      kind: PersistentVolumeClaim
      metadata:
        name: mydata
      spec:
        accessModes:
        - ReadWriteOnce
        resources:
          requests:
            storage: 10Mi
` 
I got the following error 
`Error from server (BadRequest): error when creating "test-broker.yaml": ActiveMQArtemis in version "v1beta1" cannot be handled as a ActiveMQArtemis: strict decoding error: unknown field "spec.deploymentPlan.extraVolumeClaimTemplates[0].metadata.name"
`
Wondering what's the correct way to configure the pvc's name.

Reply:
@gaohoward I think this is an issue of the kustomize tool. We could create the `VolumeClaimTemplate` struct to emulate the array of [PersistentVolumeClaim](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#persistentvolumeclaim-v1-core) as in [statefulset](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#statefulset-v1-apps), i.e.
```
...
	// Specifies Extra Volume Claims Templates for the broker pods
	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Extra Volume Claims Templates"
	ExtraVolumeClaimTemplates []VolumeClaimTemplate `json:"extraVolumeClaimTemplates,omitempty"`
}

type VolumeClaimTemplate struct {
	// Standard object's metadata.
	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Metadata"
	ObjectMeta `json:"metadata,omitempty"`

	// spec defines the desired characteristics of a volume requested by a pod author.
	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#persistentvolumeclaims
	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Spec"
	Spec corev1.PersistentVolumeClaimSpec `json:"spec,omitempty"`
}

type ObjectMeta struct {
	// Name must be unique within a namespace. Is required when creating resources, although
	// some resources may allow a client to request the generation of an appropriate name
	// automatically. Name is primarily intended for creation idempotence and configuration
	// definition.
	// Cannot be updated.
	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names#names
	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Name"
	Name string `json:"name,omitempty"`

	// Namespace defines the space within which each name must be unique. An empty namespace is
	// equivalent to the "default" namespace, but "default" is the canonical representation.
	// Not all objects are required to be scoped to a namespace - the value of this field for
	// those objects will be empty.
	//
	// Must be a DNS_LABEL.
	// Cannot be updated.
	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces
	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Namespace"
	Namespace string `json:"namespace,omitempty"`

	// Annotations is an unstructured key value map stored with a resource that may be
	// set by external tools to store and retrieve arbitrary metadata. They are not
	// queryable and should be preserved when modifying objects.
	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations
	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Annotations"
	Annotations map[string]string `json:"annotations,omitempty"`

	// Map of string keys and values that can be used to organize and categorize
	// (scope and select) objects. May match selectors of replication controllers
	// and services.
	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels
	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Labels"
	Labels map[string]string `json:"labels,omitempty"`
}
```

Reply:
+1